### PR TITLE
KFSPTS-13518 modify create accounting document job so that it fails on non business rule failures

### PR DESCRIPTION
**Pull request by jhulslander**

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/fp/batch/CreateAccountingDocumentsStep.java [hunk 1]
```diff
@@ -2,18 +2,29 @@
...
 
     @Override
     public boolean execute(String jobName, Date jobRunDate) throws InterruptedException {
-        createAccountingDocumentService.createAccountingDocumentsFromXml();
-        return true;
+        boolean processResults;
+        try {
+            processResults = createAccountingDocumentService.createAccountingDocumentsFromXml();
+        } catch (Exception e) {
+            LOG.error("execute, An error has ocurred while processing input files.", e);
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/fp/batch/CreateAccountingDocumentsStep.java]: changed

File: src/main/java/edu/cornell/kfs/fp/batch/CreateAccountingDocumentsStep.java [hunk 2]
```diff
@@ -2,18 +2,29 @@
...
+            processResults = false;
+        }
+        LOG.info("execute, the processing results are: " + processResults);
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/fp/batch/CreateAccountingDocumentsStep.java]: reworked

File: src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java [hunk 3]
```diff
@@ -52,17 +52,23 @@
...
-    public void createAccountingDocumentsFromXml() {
+    public boolean createAccountingDocumentsFromXml() {
         List<String> inputFileNames = batchInputFileService.listInputFileNamesWithDoneFile(accountingDocumentBatchInputFileType);
         LOG.info("createAccountingDocumentsFromXml: Found " + inputFileNames.size() + " files to process");
         
+        CreateAccountingDocumentLogReport logReport = new CreateAccountingDocumentLogReport();
+        
         inputFileNames.stream()
-                .forEach(this::processAccountingDocumentFromXml);
+                .forEach(fileName -> {processAccountingDocumentFromXml(fileName, logReport);});
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java]: good catch

File: src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java [hunk 4]
```diff
@@ -52,17 +52,23 @@
...
         
+        LOG.info("createAccountingDocumentsFromXml, files with non business rule errors: " + logReport.getFilesWithNonBusinessRuleFailures());
+        LOG.info("createAccountingDocumentsFromXml, files without non business rule errors: " + logReport.getFilesWithoutNonBusinessRuleFailures());
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java]: changed

File: src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java [hunk 5]
```diff
@@ -836,24 +880,25 @@ protected String buildGeneratorBeanName(AccountingDocumentGenerator<?> documentG
...
             UserSession systemUserSession = MockPersonUtil.createMockUserSession(systemUser);
-            
+
+            boolean result = false;
             try {
-                GlobalVariables.doInNewGlobalVariables(systemUserSession, () -> {
-                    super.createAccountingDocumentsFromXml();
-                    return null;
+            	result = GlobalVariables.doInNewGlobalVariables(systemUserSession, () -> {
+                	return super.createAccountingDocumentsFromXml();
```
[the remark this answers is not in the document]
Comment (cah292) [reply src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java]: Ok, leaving it like this for readability is fine. Thanks for the info.

File: src/main/java/edu/cornell/kfs/fp/batch/CreateAccountingDocumentsStep.java [hunk 6]
```diff
@@ -2,18 +2,33 @@
...
+        try {
+            processResults = createAccountingDocumentService.createAccountingDocumentsFromXml();
+        } catch (Exception e) {
+            LOG.error("execute, An error has occurred while processing input files.", e);
+            processResults = false;
+        }
+        if (processResults) {
+            LOG.info("execute, The job ran successfully");
+        } else {
+            LOG.info("execute, there non business rules errors that caused this job to fail");
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/fp/batch/CreateAccountingDocumentsStep.java]: good point about the warning 
I made the change

File: src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java [hunk 7]
```diff
@@ -231,5 +245,25 @@ public void setCreateAccountingDocumentReportService(
     public void setParameterService(ParameterService parameterService) {
         this.parameterService = parameterService;
     }
+    
+    protected class CreateAccountingDocumentLogReport {
+        private List<String> filesWithNonBusinessRuleFailures;
+        private List<String> filesSuccessfullyProccessed;
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java]: ok

File: src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java [hunk 8]
```diff
@@ -231,5 +245,25 @@ public void setCreateAccountingDocumentReportService(
...
+        public CreateAccountingDocumentLogReport() {
+            filesWithNonBusinessRuleFailures = new ArrayList<String>();
+            filesSuccessfullyProccessed = new ArrayList<String>();
+        }
+
+        public List<String> getFilesWithNonBusinessRuleFailures() {
+            return filesWithNonBusinessRuleFailures;
+        }
+
+        public List<String> getFilesSuccesssfullyProcessed() {
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java]: ok

**Comment by jhulslander** [comment]
I've made further code review changes, this is ready for review again

**Comment by jhulslander** [comment]
I made the changes specified, ready for review again

**Pull request merged**